metricbeat/module/logstash/logstash.go

```diff
@@ -80,6 +78,24 @@ type PipelineState struct {
 	Workers        int             `json:"workers"`
 }
 
+// Vertex represents a vertex in node_stats
+type Vertex struct {
```

Comment:
I was able to get the Pipeline viewer loading by adding the following properties to the struct, but we should compare `node` and `node_stats` documents generated by this change with one generated by main branch to ensure parity

```
+       Type                      string `json:"type"`
+       PluginType                string `json:"plugin_type"`
+       ConfigName                string `json:"config_name"`
```

Reply:
Yah, I was worried about something like this. It seems fairly hard to come up with an authoritative list of what can be in any one of these fields?

Reply:
If we already know the offending properties can we default to unwrap everything (like it was) and handle these properties as special cases and cast them after deserialization ?

Also, can we fix this at the mapping layer by enforcing the type of these properties as `long` in the index template ? I'm wondering if this is caused by dynamic mapping being enabled

As for the authoritative list, I think we can rely on the respective API responses but the logstash folks may have that defined somewhere in code

Reply:
> Also, can we fix this at the mapping layer by enforcing the type of these properties as long in the index template, ES should make these cast for us right ? I'm wondering if this is caused by dynamic mapping being enabled

So, a number of the problem fields are already defined properly in the mapping: https://github.com/elastic/beats/blob/main/metricbeat/module/logstash/node_stats/_meta/fields.yml

The issue seems to be when events are sent to ES via logstash, which seems to interpret types from context, hence ES sometimes getting floats or long values.

That being said, I do kind of wonder if there's a better way to do this, as hard-defining the values seems to be a tad risky.

Reply:
That sounds odd to me, if the field is properly mapped there should not be any issues and ES would silently coerce the value right ? [The original issue happened on 7.16](https://github.com/elastic/beats/blob/7.16/metricbeat/module/logstash/node_stats/_meta/fields.yml) and the fields were not mapped on this version https://github.com/elastic/beats/blob/7.16/metricbeat/module/logstash/node_stats/_meta/fields.yml, so they would be dynamically mapped and explain the float to long problem. Do we have occurrences of this issue on recent versions ?

Reply:
> Do we have occurrences of this issue on recent versions ?

Yes.  There is an active, resent support case involving Logstash + Metricbeat v8.2.3

Reply:
Case #00985077
Lots of convincing screenshots.

Reply:
> which implies that the field is mapped to float in the first place, but the mappings are defining the field as long. 

@klacabane so, this is only happening on fields that have enough trialing digits that when metricbeat reads it as a float, it'll turn it into something like `6.43453e10` or whatever, which is where things are choking when we later read it as a long. Assumed that was a logstash-meets-mapping conflict, but maybe it's ES too?